Mathlib/Data/Fintype/Units.lean

```diff
@@ -36,11 +37,18 @@ instance [Monoid α] [Fintype α] [DecidableEq α] : Fintype αˣ :=
 
 instance [Monoid α] [Finite α] : Finite αˣ := Finite.of_injective _ Units.ext
 
-theorem Fintype.card_units [GroupWithZero α] [Fintype α] [Fintype αˣ] :
-    Fintype.card αˣ = Fintype.card α - 1 := by
+theorem Fintype.card_units_add_one [GroupWithZero α] [Fintype α] [Fintype αˣ] :
+    Fintype.card α = Fintype.card αˣ + 1 := by
   classical
-    rw [eq_comm, Nat.sub_eq_iff_eq_add (Fintype.card_pos_iff.2 ⟨(0 : α)⟩),
-      Fintype.card_congr (unitsEquivNeZero α)]
-    have := Fintype.card_congr (Equiv.sumCompl (· = (0 : α))).symm
+    rw [eq_comm, Fintype.card_congr (unitsEquivNeZero α)]
+    have := Fintype.card_congr (Equiv.sumCompl (· = (0 : α)))
     rwa [Fintype.card_sum, add_comm, Fintype.card_subtype_eq] at this
+
+theorem Fintype.nat_card_units_add_one [GroupWithZero α] [Fintype α] [Fintype αˣ] :
+    Nat.card α = Nat.card αˣ + 1 := by
+  rw [Nat.card_eq_fintype_card, Nat.card_eq_fintype_card, Fintype.card_units_add_one]
```

Comment:
You should use either `Fintype` and `Fintype.card` (old) or `Finite` and `Nat.card` (new) but not a mixture. So this should be:
```suggestion
theorem Nat.card_eq_card_units_add_one [GroupWithZero α] [Finite α] :
    Nat.card α = Nat.card αˣ + 1 := by
  have : Fintype α := Fintype.ofFinite α
  classical
    rw [Nat.card_eq_fintype_card, Nat.card_eq_fintype_card, Fintype.card_eq_card_units_add_one]
```

Reply:
Ok got it, thanks!